### PR TITLE
V3 ToF support

### PR DESCRIPTION
**Pull request by SzabolcsGergely**

### CODE REVIEW

File: include/depthai/pipeline/datatype/ImgFrame.hpp [hunk 1]
```diff
@@ -17,6 +16,7 @@
 
 // optional
 #ifdef DEPTHAI_HAVE_OPENCV_SUPPORT
+    #include <opencv2/core/mat.hpp>
```
Comment (SzabolcsGergely) [inline include/depthai/pipeline/datatype/ImgFrame.hpp:19]: Maybe adding a hil job to compile w/o DEPTHAI_HAVE_OPENCV_SUPPORT would be useful.

File: bindings/python/src/pipeline/datatype/ToFConfigBindings.cpp [hunk 2]
```diff
@@ -34,35 +31,23 @@ void bind_tofconfig(pybind11::module& m, void* pCallstack){
...
     toFConfig
         .def(py::init<>())
         // .def(py::init<std::shared_ptr<ToFConfig>>())
-        
-        .def("setDepthParams", static_cast<ToFConfig&(ToFConfig::*)(dai::ToFConfig::DepthParams)>(&ToFConfig::setDepthParams), py::arg("config"), DOC(dai, ToFConfig, setDepthParams))
-        .def("setFreqModUsed", static_cast<ToFConfig&(ToFConfig::*)(dai::ToFConfig::DepthParams::TypeFMod)>(&ToFConfig::setFreqModUsed), DOC(dai, ToFConfig, setFreqModUsed))
-        .def("setAvgPhaseShuffle", &ToFConfig::setAvgPhaseShuffle, DOC(dai, ToFConfig, setAvgPhaseShuffle))
-        .def("setMinAmplitude", &ToFConfig::setMinAmplitude, DOC(dai, ToFConfig, setMinAmplitude))
+
+        .def_readwrite("median", &ToFConfig::median, DOC(dai, ToFConfig, median))
```
Comment (SzabolcsGergely) [inline bindings/python/src/pipeline/datatype/ToFConfigBindings.cpp:39]: we could rename to medianFilter

File: src/pipeline/node/ToF.cpp [hunk 3]
```diff
@@ -12,5 +12,63 @@ ToF::Properties& ToF::getProperties() {
...
+                        }
+                    }
+                }
+            }
+            if(found) {
+                boardSocket = socket;
+                break;
+            }
+        }
+    }
```
Comment (moratom) [inline src/pipeline/node/ToF.cpp:44]: Could we add the FPS here as well?

File: include/depthai/properties/ImageAlignProperties.hpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+struct ImageAlignProperties : PropertiesSerializable<Properties, ImageAlignProperties> {
+    ImageAlignConfig initialConfig;
+
+    /// Num frames in output pool
+    int numFramesPool = 4;
+
+    /**
+     * Optional output width
+     */
+    int alignWidth = 0;
```
[the remark this answers is not in the document]
Comment (SzabolcsGergely) [reply include/depthai/properties/ImageAlignProperties.hpp:24]: Oh, my bad, I thought it's ToFProperties. It's implemented for ImageAlign

**Pull request merged**